modVM/src/org/aion/vm/ExecutionBatch.java

```diff
@@ -127,10 +127,7 @@ private KernelTransactionContext constructTransactionContext(
         long energyLimit = transaction.nrgLimit();
         long nrg = transaction.nrgLimit() - transaction.transactionCost(block.getNumber());
         DataWord callValue = new DataWord(ArrayUtils.nullToEmpty(transaction.getValue()));
-        byte[] callData =
-                transaction.isContractCreationTransaction()
-                        ? ByteUtil.EMPTY_BYTE_ARRAY
-                        : ArrayUtils.nullToEmpty(transaction.getData());
+        byte[] callData = ArrayUtils.nullToEmpty(transaction.getData());
```

Comment:
This is because of our close coupling, we didn't realize that this conditional was an fvm implementation detail and not a high level concept. 

This batch sends the contexts off to the VM's, and the AVM relies on this field being set as the actual data. The FVM, however, does not rely on this, and it made this conditional choice to anticipate its later serialization concerns, which the AVM does not have.

So at this level we should be passing the actual data. In the contexts that the FVM uses, however, we need to preserve this logic for the `toBytes()`